modules/custom/social_gdpr/social_gdpr.module

```diff
@@ -49,6 +51,84 @@ function social_gdpr_form_user_register_form_alter(&$form, FormStateInterface $f
   }
 }
 
+/**
+ * Implements hook_form_FORM_ID_alter() for "user_form".
+ *
+ * Make possible to revoke or give consent of policies for user.
+ */
+function social_gdpr_form_user_form_alter(array &$form, FormStateInterface $form_state): void {
+  // Do not apply for register form.
+  if ($form['#form_id'] === 'user_register_form') {
+    return;
+  }
+
+  // "Data policies" are build by data policy module on user register form.
+  // See data_policy_form_user_register_form_alter().
+  // The alter function perform a "data policy" form element build and
+  // do all other stuff (submit, validating etc.).
+  // Unfortunately, we can't just reuse the function, because it adds submit
+  // handler that will add new consents rather than edit existed.
```

Comment:
Is there possibility of improving this in https://www.drupal.org/project/data_policy module itself?

Reply:
I kind of agree, would be valuable and would avoid maybe using the form_alter which I thought we wanted to avoid if possible, as later with decoupled we can't use this.

Reply:
Hi guys, @nechai is on off this week, he will be back next week and then should be able to check it.

Reply:
Hey @ribel I'm atm already checking it out to see what we can do. Maybe I will change the data-policy module itself with some code from here.

Reason is that currently the addcheckbox module takes currentuser into account, so to fix a bug I would already need to release a new version of the data_policy module. So maybe I can partially port this code there anyway.

Reply:
Hi @robertragas,
That would be great, since we anyway don't have maintainer access to data_policy module.